main.tf

```diff
@@ -49,60 +48,74 @@ resource "aws_cloudwatch_log_group" "control_plane" {
   tags              = var.tags
   kms_key_id        = local.kms_cmk_arn
 }
-
 /*
   Allow nodes to join the cluster
 */
 
-data "aws_iam_role" "node_role" {
-  name = var.iam_config.node_role
-}
-
 locals {
-  aws_auth_role_map = concat(
-    [
-      {
-        rolearn  = data.aws_iam_role.node_role.arn
-        username = "system:node:{{EC2PrivateDNSName}}"
-        groups   = ["system:bootstrappers", "system:nodes"]
-      },
-    ],
-    var.aws_auth_role_map,
-  )
+  aws_auth_configmap_data = {
+    mapRoles = yamlencode(concat(
+      [
+        {
+          rolearn  = aws_iam_role.karpenter_fargate.arn
+          username = "system:node:{{SessionName}}"
+          groups = [
+            "system:bootstrappers",
+            "system:nodes",
+            "system:node-proxier",
+          ]
+        },
+        {
+          rolearn  = aws_iam_role.karpenter_node.arn
```

Comment:
In web-platform-X  we use` instanceProfile: EKSNode` [here](https://github.com/cookpad/global-web-platform/blob/main/k8s/infrastructure/base/karpenter/provisioner-default.yaml#L35) we will have to change it in the provisioner?

Reply:
That is the intention, but we can add the old profile name to the map during transition.